README.md

```diff
@@ -1,11 +1,81 @@
+* `class_name` (string, optional): The name of the Class in the CDK template to synthesize. Default: `CdkStack`.
+* `context` (object, optional): The context for the CDK Stack. See [CDK Context](https://docs.aws.amazon.com/cdk/v2/guide/context.html) for further info.
+
+#### CDK Feature Flags and Custom Bootstrap
+
+To set any [CDK Feature Flags](https://docs.aws.amazon.com/cdk/v2/guide/featureflags.html) or to specify a modified
+[CDK Bootstrap](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html) qualifier, set these in the handler's
+`context` argument. See [sceptre-example](sceptre-example) for an example of this.
+
+Reminder: the `context` argument is a standard Sceptre property, so resolvers and/or Jinja variables can be used in the value. 
+
+### How does this handler work?
+
+When using _only_ the CDK CLI (not Sceptre) to deploy using `cdk deploy`, the CDK CLI effectively performs
+the following steps:
+
+1. Synthesises any constructs defined within the CDK Class into a CloudFormation Template and a
+bundle of required assets.
+2. Publishes the assets to a CDK bootstrapped S3 bucket and/or ECR registry.
+3. Creates/updates a CloudFormation stack from the synthesized template.
+
+When you use Sceptre with this handler, the CDK handler performs steps 1-2 above to create a template
+that Sceptre can use, **but it does not use CDK to deploy it!**. Instead, Sceptre can use that template
+produced in step 1 above to perform all it's usual commands with all it's usual magic!
+
+In other words, using this handler lets you use resolvers, put your CDK stack into StackGroups, let
+you name your stack according to Sceptre's naming conventions, `validate`, `diff`, and more! Basically,
+the CDK stack can be managed using Sceptre just like any other.
+
+### IAM and authentication
+
+This handler uses the stack's connection information to generate AWS environment variables and sets
+those on the CDK process, ensuring that the AWS authentication configuration on the stack config and
+project is carried over to CDK without any need for additional arguments.
+
+**Important:** CDK creates CloudFormation-ready templates and uses `cdk_assets` to publish artifacts
+to S3 and ECR in the process. This means that Sceptre commands that do not normally require S3 and ECR
+actions (such as `generate`, `validate`, `diff`, and others) will require them when using this
+handler. You will need to ensure that any user or role executing these commands has proper
+permissions for these operations.
+
+### Sceptre Management of the CDK Bootstrap
+
+To optionally manage the CDK bootstrap CloudFormation template and stack with Sceptre, the bootstrap
+template can be generated using the AWS CDK CLI: `cdk bootstrap --show-template > cdk-bootstrap.yaml`.
+This can be deployed into a stack using the standard Sceptre process. Note: the stack needs to be called
+`CDKToolkit` so the `stack_name` property of the Sceptre config needs to specify this. 
```

Comment:
> Note: the stack needs to be called `CDKToolkit` so the `stack_name` property of the Sceptre config needs to specify this. 

This is a little confusing to me. What I _think_ you're referring to is the CDK stack _class_ name and the `class_name` option. But maybe you're really referring to the CloudFormation stack and that it HAS to be called "CDKToolkit". If that's true... I thought that cdk bootstraps the stack names with some weird alphanumeric prefix or suffix or something. I'm pretty sure it's possible to have multiple bootstrapped stacks. Anyway, I'm pretty confused. And if I'm confused, I'm thinking others might too.